In [9]:
from google.colab import drive
drive.mount('/content/drive/')

Drive already mounted at /content/drive/; to attempt to forcibly remount, call drive.mount("/content/drive/", force_remount=True).


In [10]:
%cd /content/drive/MyDrive/Colab Notebooks/project/UNET
%ls

/content/drive/MyDrive/Colab Notebooks/project/UNET
 before_TF_run3_checkpoint.pt   GPU_run_checkpoint_1.pth.tar   TF_run4_checkpoint.pt
 before_TF_run4_checkpoint.pt  'Kopi af golf_utils.py'         TF_run5_checkpoint.pt
 before_TF_run5_checkpoint.pt   metrics.csv                    TF_run6_checkpoint.pt
 before_TF_run6_checkpoint.pt   __pycache__/                   thickBowl_Run_6_GPU.ipynb
 checkpoint.pt                  run2_checkpoint.pt             thickPlate_Run_4_GPU.ipynb
 datasetHandler.py              Run_2_GPU.ipynb                thinBowl_Run_5_GPU.ipynb
 golfHandler.py                 run3_checkpoint.pt             thinPlate_Run_3_GPU.ipynb
 golf_predict_gpu.py            run4_checkpoint.pt             UNET_model_OG.ipynb
 golf_predict.py                run5_checkpoint.pt             unetSimple.py
 golf_train.py                  run6_checkpoint.pt             utils_temp.txt
 golf_utils.py                  TF_run3_checkpoint.pt


In [ ]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 1993
Number of masks: 1993
=> Loading checkpoint
Model loadet
Let's predict
Starting prediction...
Evaluating: 100% 1993/1993 [11:15<00:00,  2.95it/s]
Average Dice score: 0.9617909257552532
Average IoU score: 0.9289590279925811
Average Precision: 0.9691054377935838
Average Recall: 0.9568972998855796
Average F1 score: 0.9617909207600769
Average Accuracy: 0.9926182262920213
Average MAE: 0.007381773541171648
Average MSE: 0.007381773541171648
Average Dice Loss: 0.03820907381749668
Average BCE Loss: 0.02956431018206353
Average FPS: 66.71764645064864
Average Prediction Time per Image: 0.005159188561025601
Total Prediction Time for Epoch: 10.282262802124023
Test Dice Score: 0.9618
Test IoU Score: 0.9290
Test Precision: 0.9691
Test Recall: 0.9569
Test F1 Score: 0.9618
Test Accuracy: 0.9926
Test MAE: 0.0074
Test MSE: 0.0074
Test Dice Loss: 0.0382
Test BCE Loss: 0.0296
Test FPS: 66.72
Test

# Second prediction

In [ ]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 1993
Number of masks: 1993
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 1993/1993 [01:17<00:00, 25.69it/s]
Average Dice score: 0.9630378433279674
Average IoU score: 0.9319600625242248
Average Precision: 0.9633154335376125
Average Recall: 0.9656390796125771
Average F1 score: 0.9630378383343039
Average Accuracy: 0.9929391346274461
Average MAE: 0.007060865217743173
Average MSE: 0.007060865217743173
Average Dice Loss: 0.0369621561629893
Average BCE Loss: 0.032428997382802006
Average FPS: 39.807146887707006
Average Prediction Time per Image: 0.00652638967469058
Total Prediction Time for Epoch: 13.007094621658325
Test Dice Score: 0.9630
Test IoU Score: 0.9320
Test Precision: 0.9633
Test Recall: 0.9656
Test F1 Score: 0.9630
Test Accuracy: 0.9929
Test MAE: 0.0071
Test MSE: 0.0071
Test Dice Loss: 0.0370
Test BCE Loss: 0.0324
Test FPS: 39.81
Test

# Thick Bowl new FPS



In [4]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 1993
Number of masks: 1993
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 1993/1993 [01:31<00:00, 21.69it/s]
Average Dice score: 0.9630378433279674
Average IoU score: 0.9319600625242248
Average Precision: 0.9633154335376125
Average Recall: 0.9656390796125771
Average F1 score: 0.9630378383343039
Average Accuracy: 0.9929391346274461
Average MAE: 0.007060865217743173
Average MSE: 0.007060865217743173
Average Dice Loss: 0.0369621561629893
Average BCE Loss: 0.032428997382802006
Average FPS: 284.31419463615464
Average Prediction Time per Image: 0.003517235575521405
Total Prediction Time for Epoch: 7.00985050201416
Test Dice Score: 0.9630
Test IoU Score: 0.9320
Test Precision: 0.9633
Test Recall: 0.9656
Test F1 Score: 0.9630
Test Accuracy: 0.9929
Test MAE: 0.0071
Test MSE: 0.0071
Test Dice Loss: 0.0370
Test BCE Loss: 0.0324
Test FPS: 284.31
Test

# TRACKMAN FPS FIX

In [12]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 112
Number of masks: 112
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 112/112 [00:24<00:00,  4.62it/s]
Average Dice score: 0.45160199125586065
Average IoU score: 0.30618109529509396
Average Precision: 0.3180700197321778
Average Recall: 0.9340833767847164
Average F1 score: 0.4516019877251708
Average Accuracy: 0.9300434716710756
Average MAE: 0.06995652835550052
Average MSE: 0.06995652835550052
Average Dice Loss: 0.5483980067074299
Average BCE Loss: 0.17459109498720085
Average FPS: 147.67617512326203
Average Prediction Time per Image: 0.006771572998591832
Total Prediction Time for Epoch: 0.7584161758422852
Test Dice Score: 0.4516
Test IoU Score: 0.3062
Test Precision: 0.3181
Test Recall: 0.9341
Test F1 Score: 0.4516
Test Accuracy: 0.9300
Test MAE: 0.0700
Test MSE: 0.0700
Test Dice Loss: 0.5484
Test BCE Loss: 0.1746
Test FPS: 147.68
Test Av

# Testing on 112 images from other dataset of golfers

270x480

In [ ]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 112
Number of masks: 112
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 112/112 [00:17<00:00,  6.25it/s]
Average Dice score: 0.45160199125586065
Average IoU score: 0.30618109529509396
Average Precision: 0.3180700197321778
Average Recall: 0.9340833767847164
Average F1 score: 0.4516019877251708
Average Accuracy: 0.9300434716710756
Average MAE: 0.06995652835550052
Average MSE: 0.06995652835550052
Average Dice Loss: 0.5483980067074299
Average BCE Loss: 0.17459109498720085
Average FPS: 11.602627068724988
Average Prediction Time per Image: 0.010504875864301409
Total Prediction Time for Epoch: 1.1765460968017578
Test Dice Score: 0.4516
Test IoU Score: 0.3062
Test Precision: 0.3181
Test Recall: 0.9341
Test F1 Score: 0.4516
Test Accuracy: 0.9300
Test MAE: 0.0700
Test MSE: 0.0700
Test Dice Loss: 0.5484
Test BCE Loss: 0.1746
Test FPS: 11.60
Test Ave

# Testing on 112 images from other dataset of golfers

160x160 testimages4

In [ ]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 112
Number of masks: 112
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 112/112 [01:30<00:00,  1.24it/s]
Average Dice score: 0.7515390076990631
Average IoU score: 0.6662454813219634
Average Precision: 0.6898340213384963
Average Recall: 0.8408441484009875
Average F1 score: 0.7515390033823113
Average Accuracy: 0.9801335797991069
Average MAE: 0.01986641963178824
Average MSE: 0.01986641963178824
Average Dice Loss: 0.23953242067779815
Average BCE Loss: 0.06671821749270228
Average FPS: 22.005017062058226
Average Prediction Time per Image: 0.010677014078412737
Total Prediction Time for Epoch: 1.1958255767822266
Test Dice Score: 0.7515
Test IoU Score: 0.6662
Test Precision: 0.6898
Test Recall: 0.8408
Test F1 Score: 0.7515
Test Accuracy: 0.9801
Test MAE: 0.0199
Test MSE: 0.0199
Test Dice Loss: 0.2395
Test BCE Loss: 0.0667
Test FPS: 22.01
Test Aver

Second try

In [ ]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 112
Number of masks: 112
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 112/112 [00:11<00:00,  9.37it/s]
Average Dice score: 0.7515390076990631
Average IoU score: 0.6662454813219634
Average Precision: 0.6898340213384963
Average Recall: 0.8408441484009875
Average F1 score: 0.7515390033823113
Average Accuracy: 0.9801335797991069
Average MAE: 0.01986641963178824
Average MSE: 0.01986641963178824
Average Dice Loss: 0.23953242067779815
Average BCE Loss: 0.06671821749270228
Average FPS: 38.11728006934043
Average Prediction Time per Image: 0.008840684379850115
Total Prediction Time for Epoch: 0.9901566505432129
Test Dice Score: 0.7515
Test IoU Score: 0.6662
Test Precision: 0.6898
Test Recall: 0.8408
Test F1 Score: 0.7515
Test Accuracy: 0.9801
Test MAE: 0.0199
Test MSE: 0.0199
Test Dice Loss: 0.2395
Test BCE Loss: 0.0667
Test FPS: 38.12
Test Avera

A100 try

In [ ]:
!python golf_predict_gpu.py

Number of images: 9196
Number of masks: 9196
Number of images: 1955
Number of masks: 1955
Number of images: 112
Number of masks: 112
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 112/112 [00:06<00:00, 17.43it/s]
Average Dice score: 0.75156082856313
Average IoU score: 0.6662801320418756
Average Precision: 0.6898747107382396
Average Recall: 0.8408424554598748
Average F1 score: 0.7515608242463662
Average Accuracy: 0.9801356724330358
Average MAE: 0.019864327074693783
Average MSE: 0.019864327074693783
Average Dice Loss: 0.23951059952378273
Average BCE Loss: 0.06671152208998267
Average FPS: 81.87295560325182
Average Prediction Time per Image: 0.006256188665117536
Total Prediction Time for Epoch: 0.7006931304931641
Test Dice Score: 0.7516
Test IoU Score: 0.6663
Test Precision: 0.6899
Test Recall: 0.8408
Test F1 Score: 0.7516
Test Accuracy: 0.9801
Test MAE: 0.0199
Test MSE: 0.0199
Test Dice Loss: 0.2395
Test BCE Loss: 0.0667
Test FPS: 81.87
Test Avera

# Transfer Learning

In [ ]:
!python golf_train.py

Number of images: 70
Number of masks: 70
Number of images: 30
Number of masks: 30
Number of images: 12
Number of masks: 12
Print loading model...
=> Loading checkpoint
Model loaded
Let's train
Epoch 1/100
100% 70/70 [00:18<00:00,  3.83it/s, loss=0.00775]
Evaluating: 100% 30/30 [00:10<00:00,  2.85it/s]
Average Dice score: 0.7528412943782777
Average IoU score: 0.641358671102494
Average Precision: 0.8399284965972804
Average Recall: 0.7160427110365872
Average F1 score: 0.7528412897369635
Average Accuracy: 0.9848032407407407
Average MAE: 0.01519675924209878
Average MSE: 0.01519675924209878
Average Dice Loss: 0.2471587061882019
Average BCE Loss: 0.04214847505403062
Average FPS: 5.891642649029916
Average Prediction Time per Image: 0.004117639859517416
Total Prediction Time for Epoch: 0.12352919578552246
Current learning rate remains at 0.001
=> Saving checkpoint
Epoch [1/100]
Validation Dice Score: 0.7528
Validation IoU Score: 0.6414
Validation Precision: 0.8399
Validation Recall: 0.7160
Vali

In [ ]:
!python golf_predict_gpu.py

Number of images: 70
Number of masks: 70
Number of images: 30
Number of masks: 30
Number of images: 12
Number of masks: 12
=> Loading checkpoint
Model loaded
Let's predict
Starting prediction...
Evaluating: 100% 12/12 [00:06<00:00,  1.80it/s]
Average Dice score: 0.8755675742639945
Average IoU score: 0.7960943340367265
Average Precision: 0.9129558389913187
Average Recall: 0.8667460647505867
Average F1 score: 0.87556756936686
Average Accuracy: 0.9937763631687243
Average MAE: 0.006223636766662821
Average MSE: 0.006223636766662821
Average Dice Loss: 0.12443243463834126
Average BCE Loss: 0.021094530316380162
Average FPS: 8.575892631227052
Average Prediction Time per Image: 0.02535271644592285
Total Prediction Time for Epoch: 0.3042325973510742
Test Dice Score: 0.8756
Test IoU Score: 0.7961
Test Precision: 0.9130
Test Recall: 0.8667
Test F1 Score: 0.8756
Test Accuracy: 0.9938
Test MAE: 0.0062
Test MSE: 0.0062
Test Dice Loss: 0.1244
Test BCE Loss: 0.0211
Test FPS: 8.58
Test Average Prediction